encodings/runend/src/runend.rs

```diff
@@ -84,6 +84,23 @@ impl RunEndArray {
             .map(|s| s.to_ends_index(self.ends().len()))
     }
 
+    /// Convert a batch of logical indices into an index for the values.
+    ///
+    /// See: [find_physical_index][Self::find_physical_index].
+    pub fn find_physical_indices(&self, indices: &[u64]) -> VortexResult<Vec<usize>> {
+        search_sorted_u64_many(
+            &self.ends(),
+            indices,
+            &vec![SearchSortedSide::Right; indices.len()],
+        )
+        .map(|results| {
+            results
+                .iter()
+                .map(|result| result.to_ends_index(self.ends().len()))
+                .collect()
```

Comment:
this should not allocate and occur in-place

Reply:
If results is dropped after this then yes this will usually be the case.